pkg/services/internal/whatsonchain/woc_get_merkle_path.go

```diff
@@ -0,0 +1,167 @@
+package whatsonchain
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/wdk"
+	"github.com/bsv-blockchain/go-sdk/chainhash"
+	"github.com/bsv-blockchain/go-sdk/transaction"
+	"github.com/go-resty/resty/v2"
+	"github.com/go-softwarelab/common/pkg/must"
+	"github.com/go-softwarelab/common/pkg/to"
+)
+
+// tscProof represents the response from the WoC /tx/{txid}/proof/tsc endpoint
+type tscProof struct {
+	Index  int      `json:"index"`
+	Nodes  []string `json:"nodes"`
+	Target string   `json:"target"` // block hash
+	TxOrID string   `json:"txOrId"` // txid
+}
+
+// blockHeaderResponse represents the response from WoC /block/{hash}/header
+type blockHeaderResponse struct {
+	Height     int    `json:"height"`
+	MerkleRoot string `json:"merkleroot"`
+}
+
+// MerklePath retrieves the merkle path for a transaction using WoC TSC proof.
+func (woc *WhatsOnChain) MerklePath(ctx context.Context, txID string) (*wdk.MerklePathResult, error) {
+	var proof tscProof
+	req := woc.httpClient.R().
+		SetContext(ctx).
+		SetResult(&proof).
+		AddRetryCondition(retryOnTooManyRequestsStatus)
+
+	res, err := req.Get(fmt.Sprintf("%s/tx/%s/proof/tsc", woc.url, txID))
+	if err != nil {
+		return nil, fmt.Errorf("failed to query TSC proof: %w", err)
+	}
+	if res.StatusCode() == http.StatusNotFound {
+		return &wdk.MerklePathResult{Name: ServiceName}, nil
+	}
+	if res.StatusCode() != http.StatusOK {
+		return nil, fmt.Errorf("unexpected status code %d fetching TSC proof", res.StatusCode())
+	}
+	if len(res.Body()) == 0 {
+		return nil, errors.New("empty response body fetching TSC proof")
+	}
+
+	if err := json.Unmarshal(res.Body(), &proof); err != nil {
+		return nil, fmt.Errorf("failed to parse TSC proof: %w", err)
+	}
```

Comment:
You applied our suggestion but didn't remove manual json parsing
```suggestion
	var proof tscProof
	req := woc.httpClient.R().
		SetContext(ctx).
		SetResult(&proof).
		AddRetryCondition(retryOnTooManyRequestsStatus)

	res, err := req.Get(fmt.Sprintf("%s/tx/%s/proof/tsc", woc.url, txID))
	if err != nil {
		return nil, fmt.Errorf("failed to query TSC proof: %w", err)
	}
	if res.StatusCode() == http.StatusNotFound {
		return &wdk.MerklePathResult{Name: ServiceName}, nil
	}
	if res.StatusCode() != http.StatusOK {
		return nil, fmt.Errorf("unexpected status code %d fetching TSC proof", res.StatusCode())
	}
```